content/components/text-input.mdx

```diff
@@ -0,0 +1,161 @@
+
+**Required indicator:** Indicates that a value is required. Must be shown for any required field, even if all fields in the form are required.
+
+**Input (required):** Accepts user free-form text input
+
+**Leading visual:** An icon or short piece of text to reinforce the purpose of the input. Appears before the input's value.
+
+**Trailing visual:** The same as a leading visual, but appears after the input's value.
+
+**Trailing action:** An action that affects the input. Most commonly used for a button that clears the input.
+
+**Caption:** Provides additional context about the field to help users fill in the correct data or explain how the data will be used. Caption text should be as short as possible. Caption text may be displayed at the same time as a validation message, or it may be hidden if it only provides redundant information.
+
+**Validation message:** A single validation message may be displayed to provide helpful information for a user to complete their task. It is most commonly used to explain why a value is invalid so they can correct it and submit the form.
+
+## Options
+
+### Sizes
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224445197-4b71262c-e1cc-4495-88d5-e46c6474d3cc.png"
+/>
+
+A text input can be made smaller or larger depending on the context it's being used in.
+
+You could make an input smaller if it is being used in a tight space with other small elements. You could make an input larger if it's in an open space or it feels out of proportion with adjacent large elements.
+
+### Width
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224445198-7217e528-ed15-4fe9-aba9-264c7bc62e1b.png"
+/>
+
+By default, a text input will be given a default width by the browser.
+
+However, you can render a text input to fill the width of it's parent element, or set the width to whatever makes sense for your use-case. When setting a custom width, consider the length of the expected value.
+
+### Monospace
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224445199-c8f27a2b-18e3-4ac7-9764-1d32b1fdc9c7.png"
+/>
+
+The value of the text input may be rendered in a monospace font. Use this variant for inputs related to code. For example, setting a personal access token in a configuration.
+
+### With a trailing action
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224445201-7f81cf2c-992c-470b-b702-4b45d57c3234.png"
+/>
+
+A trailing action may be used to render a single action that affects the input. A trailing action is most commonly used for a button that clears the input.
+
+Ideally the trailing action button is an icon without a text label. It is also possible to render the button with a text label, but it should be avoided whenever possible.
```

Comment:
Does it make sense to duplicate a lot of this documentation from https://github.com/primer/design/pull/428/files#diff-d064802a2190980eba4e2bc71b3155f9e213d88e9c2ca7f9ae295a26f6a0a115R117 or should we link?